### PR TITLE
Package cfgen.1.0.0-preview.6

### PR DESCRIPTION
**Pull request by chris-armstrong**
### `cfgen.1.0.0-preview.6`
Generate AWS CloudFormation templates from code
This project provides the
  ability to generate AWS CloudFormation templates from OCaml code, with a support library to ease its use.
  AWS resource definitions have been used to generate constructs for building CloudFormation resources
  in a type-safe way.

---
* Homepage: https://github.com/chris-armstrong/ocaml-cfgen
* Source repo: git+https://github.com/chris-armstrong/ocaml-cfgen.git
* Bug tracker: https://github.com/chris-armstrong/ocaml-cfgen/issues

---
:camel: Pull-request generated by opam-publish v2.2.0

### CODE REVIEW

File: packages/cfgen/cfgen.1.0.0-preview.6/opam [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  ability to generate AWS CloudFormation templates from OCaml code, with a support library to ease its use.
+  AWS resource definitions have been used to generate constructs for building CloudFormation resources
+  in a type-safe way."""
+maintainer: ["Chris Armstrong <oss@chrisarmstrong.dev>"]
+authors: ["Chris Armstrong"]
+license: "MIT"
+homepage: "https://github.com/chris-armstrong/ocaml-cfgen"
+bug-reports: "https://github.com/chris-armstrong/ocaml-cfgen/issues"
+depends: [
+  "dune" {>= "3.4" & >= "3.4"}
```
Comment (raphael-proust) [inline packages/cfgen/cfgen.1.0.0-preview.6/opam:15]:
```suggestion
  "dune" {>= "3.4"}
```

Although that's "as generated by dune" if we believe the header. So maybe it has to do with the content of the dune-project file? Or something else?

It's not critical to remove anyway.

**Comment by raphael-proust** [comment]
Hiya,

The package you are proposing to be pulled has the `preview` in the version number. Is it meant to actually be pulled? Or is this a trial PR to sort things out before sending a final version?

If the latter, and considering you've opened and closed other preview PRs, it might be simpler for the future to open a single PR and keep pushing onto it. You can mark the PR as draft if you are unsure about the content yet and don't want the PR to be merged before having a chance to check the CI.

**Comment by chris-armstrong** [comment]
I was trying to get things right before i submitted PR, but I've got my self confused between the opam output, which asked me to tag my repo before submitting, and the documentation, which said to just keep pushing to the same PR. 

I found that by tagging first, i was unable to update my PR as I discovered the various blockers put in place by the CI/CD system.
It took several attempts to find the need for stricter version ranges, testing on 32 bit systems, extra lint rules on the generated opam file etc (I wish these were all documented).

I might go back and clean up my repo and submit again as it's an unhelpful version number (I wanted a prerelease version as a "Request for Comment").

File: packages/cfgen/cfgen.1.0.0-preview.6/opam (hunk 1 above)
Comment (chris-armstrong) [reply packages/cfgen/cfgen.1.0.0-preview.6/opam:15]: I added an explicit dune dependency, but it looks like it's generated this more unusual one? I am generating from a dune-project file, how do i fix this?

**Comment by raphael-proust** [comment]
> I found that by tagging first, i was unable to update my PR

Oh that's strange. Maybe it's a tooling issue (are you using `opam-publish`? something else?) or something. Anyway, thanks for the explanations.

> extra lint rules on the generated opam file

You can run `opam lint` locally to get most of the info you'd get from the linter of the CI.  
For the version constraints it's a bit harder. Especially the lower-bounds.

> I might go back and clean up my repo and submit again as it's an unhelpful version number (I wanted a prerelease version as a "Request for Comment").

Depending how wide you want to make your request for comments (how much of an audience you are looking for. You don't necessarily have to publish on opam-repository. You can instruct people to `opam pin` your package directly from (a branch or a tag of) your development repository.

Anyway. Let us know what you want to do with this PR and we can see what needs doing on our end.

**Comment by chris-armstrong** [comment]
I'm using opam publish - I may has misinterpreted the docs.

Thanks, I'll open a new one when I'm ready, I think I'll just ask people to pin from my repo for now

**Pull request closed without merging**